src/playground-project.ts

```diff
@@ -427,36 +418,67 @@ export class PlaygroundProject extends LitElement {
       }
     );
     port1.start();
-    const configureMessage: PlaygroundMessage = {
-      type: CONFIGURE_PROXY,
-      url: 'playground-service-worker.js',
-      scope: this.sandboxScope,
-      port: port2,
-    };
-    // We could constrain targetOrigin to
-    // `this._normalizedSandboxBaseUrl.origin`, but unclear if that provides any
-    // security benefit, and would add the limitation that the sandboxBaseUrl
-    // can't redirect to another origin.
-    iframeWindow.postMessage(configureMessage, '*', [configureMessage.port]);
+    this._postMessageToServiceWorkerProxy(
+      {
+        type: CONFIGURE_PROXY,
+        url: 'playground-service-worker.js',
+        scope: this.sandboxScope,
+        port: port2,
+      },
+      [port2]
+    );
   }
 
   private _onNewServiceWorkerPort(port: MessagePort) {
     const onMessage = (e: MessageEvent<PlaygroundMessage>) => {
       if (e.data.type === ACKNOWLEDGE_SW_CONNECTION) {
         port.removeEventListener('message', onMessage);
-        this._serviceWorkerAPI = wrap<ServiceWorkerAPI>(port);
-        this._serviceWorkerAPI.setFileAPI(
-          proxy({
-            getFile: (name: string) => this._getFile(name),
-          }),
-          this._sessionId
-        );
+        if (e.data.version === version) {
+          this._serviceWorkerAPI = wrap<ServiceWorkerAPI>(port);
+          this._serviceWorkerAPI.setFileAPI(
+            proxy({
+              getFile: (name: string) => this._getFile(name),
+            }),
+            this._sessionId
+          );
+        } else {
+          // Version mismatch. Request the service worker be updated
+          // immediately. We'll get back here again after it updates via a
+          // CONNECT_PROJECT_TO_SW message from the proxy.
+          console.info(
+            'Playground service worker is outdated, waiting for update.'
+          );
+          this._postMessageToServiceWorkerProxy({
+            type: UPDATE_SERVICE_WORKER,
+          });
+        }
       }
     };
     port.addEventListener('message', onMessage);
     port.start();
   }
 
+  private _postMessageToServiceWorkerProxy(
```

Comment:
nit: I didn't know what "proxy" referred to until I read the comments here. Could the name be reconciled with the `_iframe` field? Name the iframe `_proxyIFrame`, `_swProxy` or the method `_postMessageToServiceWorkerIframe`?

Reply:
Done.